### PR TITLE
Adding proper tests and increasing the API call surface

### PR DESCRIPTION
**Pull request by akitaonrails**
I wanted to add more method calls into the eth module, start the personal module and add net in the future.

Also testing a convention where user input values will be assumed to be Ether and api result values will come as Wei.

Using VCR to add real http results from a real Geth server I am maintaining. So I am sure that every API call tested does work as intended.

If this PR is accepted I intend to keep adding more of the web3 modules in the future. Dunno if there are other plans for this gem.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1 +1,8 @@
 require "bundler/gem_tasks"
+require 'rake/testtask'
+
+task :default => :test
```
Comment (veganstraightedge) [inline Rakefile:4]: This could be modified to:

`task default: :test`

File: lib/web3/eth/eth_module.rb [hunk 2]
```diff
@@ -5,47 +5,65 @@ class EthModule
 
       include Web3::Eth::Utility
 
-      PREFIX = 'eth_'
+      PREFIX = 'eth_'.freeze
 
-      def initialize web3_rpc
+      def initialize(web3_rpc)
```
Comment (veganstraightedge) [inline lib/web3/eth/eth_module.rb:10]: While the Ruby community as a whole is certainly not unified in its preference for parens or no parens in a method definition, @izetex's existing style in this project seems to be no parens. I think it's good form to maintain the same style as the project creator.

***

No parens is sometimes referred to as `Seattle.rb style`:

https://twitter.com/mrb_bk/status/429000440429031424

File: lib/web3/eth/patches.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+unless Hash.method_defined?(:compact)
```
Comment (veganstraightedge) [inline lib/web3/eth/patches.rb:1]: These look like they're all also in `activesupport`. Alternately, you could cherry pick require the bits of `activesupport` that you want.

http://guides.rubyonrails.org/active_support_core_extensions.html#cherry-picking-a-definition

File: lib/web3/eth/personal_module.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+module Web3
+  module Eth
```
Comment (veganstraightedge) [inline lib/web3/eth/personal_module.rb:2]: My personal style is to match top of the file line breaks with bottom of the file line breaks.

I'd either add some line breaks to the bottom, or remove them from the top.

**Comment by veganstraightedge** [comment]
@akitaonrails I left you a few comments on your PR. They're mostly nit-picky style suggestions. Feel free to apply them or ignore them. 👍🏻

**Comment by ToJen** [comment]
Is there an example of how this `web3-eth` would be used? Should it be a controller or helper? I want to interact with some deployed smart contracts.

**Pull request closed without merging**